### DISCUSSION TITLE
QOL for InstanceFinder registerable instances

### DISCUSSION THREAD

**Discussion by celojevic** [FirstGearGames/FishNet#254] [Feature Request]
InstanceFinder.RegisterInstance is a great and easy replacement for singletons but it is missing some QOL. One thing its missing so far is that you cant check if the instance already exists with like `InstanceFinder.HasInstance<Type>.` You _can_ do `InstanceFinder.GetInstance<Type>() != null` but there is no option to turn the warning off if the instance is not found

**Comment by FirstGearGames** [comment]
I can do that and add optional for warning(probably).

Don't forget there is also this ... `RegisterInvokeOnInstance<T>`

**Comment by FirstGearGames** [comment]
Apparently there is a warn option ... but I may have added it in after this conversion.
`public T GetInstance<T>(bool warn = true) where T : UnityComponent`

**Comment by FirstGearGames** [comment]
Added in 3.5.3
